sdk/storage/Azure.Storage.Common.DataMovement/tests/FilesystemScannerTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Storage.Common.DataMovement;
+using NUnit.Framework;
+using System.Reflection;
+
+namespace Azure.Storage.Tests
+{
+    public class FilesystemScannerTests
+    {
+        [Test]
+        public void ScanFolderContainingMixedPermissions()
+        {
+            // Arrange
+            string testPath = "C:\\Test";
```

Comment:
Let's use something like this https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettemppath?view=net-5.0&tabs=windows

Also this to create files
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=net-5.0

Reply:
Same for the other tests

Reply:
Rewrote tests to use this, all seem to pass on Windows.

TODO: Testing on Linux/macOS

I'm thinking the build errors from the pipelines at the moment might be coming from the tests. The Mono.Posix.NETStandard package I added is causing the following warning to show:
- MSB3245:	Could not resolve this reference. Could not locate the assembly "Mono.Posix". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
  - Azure.Storage.Common.DataMovement.Tests
  - Azure.Storage.Blobs.DataMovement.Tests

Windows builds fine, but since the package is only relevant for Unix/Posix platforms, it could be causing the build failures due to missing assembly. I need to set up the VM and check on this + run tests.